File: src/FsGenUI/Client/FreeSqlGeneratorClient/ClientApp/src/app/project/modals/project.ts
import { ProjectInfo } from './projectInfo';
import { GeneratorModeConfig } from './generatormodeconfig';
import { Template } from 'src/app/template/modals/template';

export class Project {
    id: number;
    projectInfo: ProjectInfo;
    projectInfoId: number;
    generatorModeConfig: GeneratorModeConfig = new GeneratorModeConfig(this.projectInfoId);
    generatorModeConfigId: number;
    builders: BuilderOptions[] = new Array<BuilderOptions>();
    globalBuilders: BuilderOptions[];
}
export class BuilderOptions {
    constructor(_type: BuilderType, _name: string) {
        this.id = 0;
        this.projectId = 0;
        this.splitDot = '_';
        this.suffix = '';
        this.templateId = 0;
        this.type = _type;
        this.isServiceOnly = false;
        this.mode = 0;
        this.name = _name;
        this.outPutPath = '';
        this.fileExtensions = 'cs';
        this.classBase = '';
        this.prefix = '';
        this.isIgnorePrefix = true;
    }
    id: number;
    projectId: number;
    project: Project;
    classBase: string;
    name: string;
    isServiceOnly: boolean;
    prefix: string;
    splitDot: string;
    isIgnorePrefix: boolean;
    outPutPath: string;
    mode: ConvertMode;
    template: Template;
    templateId: number;
    suffix: string;
    type: BuilderType;
    fileExtensions: string;
}

export enum BuilderType {
    Builder,
    GlobalBuilder
}

export enum ConvertMode {
    None,
    AllLower,
    AllUpper,
    FirstUpper
}

File: src/FreeSqlBuilder/FreeSqlBuilder.Core/Project.cs
using System.Collections.Generic;
using FreeSql.DataAnnotations;

namespace FreeSqlBuilder.Core
{
    [Table(Name = "Project")]
    public class Project : IKey<long>
    {
        [Column(IsPrimary = true, IsIdentity = true)]
        public long Id { get; set; }
        /// <summary>
        /// 项目基本信息
        /// </summary>
        public ProjectInfo ProjectInfo { get; set; }
        public long ProjectInfoId { get; set; }
        /// <summary>
        /// 生成器模式相关配置
        /// </summary>
        public GeneratorModeConfig GeneratorModeConfig { get; set; }

        public long GeneratorModeConfigId { get; set; }
        /// <summary>
        /// 多表构建器
        /// </summary>
        public List<BuilderOptions> Builders { get; set; } = new List<BuilderOptions>();
        /// <summary>
        /// 全表构建器
        /// </summary>
        public List<BuilderOptions> GlobalBuilders { get; set; } = new List<BuilderOptions>();

    }

    public interface IKey<out TKey>
    {
        [Column(IsPrimary = true, IsIdentity = true)]
        TKey Id { get; }
    }
}
